# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1344722292 14400
#      Sat Aug 11 17:58:12 2012 -0400
# Branch gui
# Node ID 098546e95a5e1b6cb8a3c2a5d878cab87a30da50
# Parent  960aa8863476bb225a82e5e12596de3688a7d409
allow location of default settings file to be configurable

* gui/default-qt-settings: Rename from default-settings.
* gui/Makefile.am (octetc_DATA): New variable.
* resource-manager.cc (default_qt_settings_file): New function.
(resource_manager::reload_settings): Call default_qt_settings_file to
find default settings file.
* run-octave.in: Set OCTAVE_DEFAULT_QT_SETTINGS in the environment for
running octave.
* gui/src/Makefile.am (liboctgui_la_CPPFLAGS): Include -I../../src,
-I../../src/interpfcn, and -I../../src/parse-tree in the list.

diff --git a/gui/Makefile.am b/gui/Makefile.am
--- a/gui/Makefile.am
+++ b/gui/Makefile.am
@@ -1,1 +1,3 @@
 SUBDIRS = src
+
+octetc_DATA = default-qt-settings
diff --git a/gui/default-settings b/gui/default-qt-settings
rename from gui/default-settings
rename to gui/default-qt-settings
diff --git a/gui/src/Makefile.am b/gui/src/Makefile.am
--- a/gui/src/Makefile.am
+++ b/gui/src/Makefile.am
@@ -87,17 +87,20 @@ liboctgui_la_CPPFLAGS = \
   -I@QT_INCDIR@/QtNetwork \
   -I$(srcdir)/../qterminal/libqterminal \
   -I$(srcdir)/m-editor \
   -I$(srcdir)/octave-adapter \
   -I$(srcdir)/qtinfo \
   -I$(srcdir)/../../libcruft/misc \
   -I../../liboctave \
   -I$(srcdir)/../../liboctave \
+  -I../../src \
   -I../../src/interp-core \
+  -I../../src/interpfcn \
+  -I../../src/parse-tree \
   -I$(srcdir)/../../src \
   -I$(srcdir)/../../src/interp-core \
   -I$(srcdir)/../../src/interpfcn \
   -I$(srcdir)/../../src/octave-value \
   -I$(srcdir)/../../src/operators \
   -I$(srcdir)/../../src/parse-tree
 
 liboctgui_la_LDFLAGS = -L@QT_LIBDIR@
diff --git a/gui/src/resource-manager.cc b/gui/src/resource-manager.cc
--- a/gui/src/resource-manager.cc
+++ b/gui/src/resource-manager.cc
@@ -10,21 +10,32 @@
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU General Public License for more details.
  *
  * You should have received a copy of the GNU General Public License
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "resource-manager.h"
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <string>
+
 #include <QFile>
 #include <QDir>
+#include <QNetworkProxy>
 
-#include <QNetworkProxy>
+#include "file-ops.h"
+#include "oct-env.h"
+
+#include "defaults.h"
+
+#include "resource-manager.h"
 
 resource_manager resource_manager::_singleton;
 
 resource_manager::resource_manager ()
 {
   _settings = 0;
   _first_run = false;
   reload_settings ();
@@ -42,28 +53,40 @@ resource_manager::get_settings ()
 }
 
 QString
 resource_manager::get_home_path ()
 {
   return _home_path;
 }
 
+static std::string
+default_qt_settings_file (void)
+{
+  std::string dsf = octave_env::getenv ("OCTAVE_DEFAULT_QT_SETTINGS");
+
+  if (dsf.empty ())
+    dsf = Voct_etc_dir + file_ops::dir_sep_str () + "default-qt-settings";
+
+  return dsf;
+}
+
 void
 resource_manager::reload_settings ()
 {
   QDesktopServices desktopServices;
   _home_path = desktopServices.storageLocation (QDesktopServices::HomeLocation);
-  QString settings_path = _home_path + "/.config/octave-gui/";
-  QString settings_file = settings_path + "settings";
+  QString settings_path = _home_path + "/.config/octave/";
+  QString settings_file = settings_path + "qt-settings";
 
   if (!QFile::exists (settings_file))
    {
      QDir("/").mkpath (settings_path);
-     QFile::copy ("../default-settings", settings_file);
+     QFile::copy (QString::fromStdString (default_qt_settings_file ()),
+                  settings_file);
      _first_run = true;
    }
   else
      _first_run = false;
 
   set_settings (settings_file);
 }
 
diff --git a/run-octave.in b/run-octave.in
--- a/run-octave.in
+++ b/run-octave.in
@@ -68,12 +68,13 @@ if [ $# -gt 0 ]; then
     shift
   elif [ "x$1" = "x-cli" ]; then
     octave_executable="$builddir/src/octave-cli"
     shift
   fi
 fi
 
 OCTAVE_SITE_INITFILE="$top_srcdir/scripts/startup/main-rcfile" \
+OCTAVE_DEFAULT_QT_SETTINGS="$top_srcdir/gui/default-qt-settings" \
   exec $builddir/libtool --mode=execute $driver \
     "$octave_executable" --no-init-path --path="$LOADPATH" \
     --image-path="$IMAGEPATH" --doc-cache-file="$DOCFILE" \
     --texi-macros-file="$TEXIMACROSFILE" --info-file="$INFOFILE" "$@"
